src/libraries/System.Private.CoreLib/generators/EventSourceGenerator.Emitter.cs

```diff
@@ -0,0 +1,109 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Text;
+using System.Threading;
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Generators
+{
+    public partial class EventSourceGenerator
+    {
+        private class Emitter
+        {
+            private StringBuilder _builder = new StringBuilder(1024);
+            private GeneratorExecutionContext _context;
+
+            public Emitter(GeneratorExecutionContext context) => _context = context;
+
+            public void Emit(EventSourceClass[] eventsources, CancellationToken cancellationToken)
+            {
+                foreach (var ec in eventsources)
+                {
+                    if (cancellationToken.IsCancellationRequested)
+                    {
+                        // stop any additional work
+                        break;
+                    }
+
+                    _builder.AppendLine("using System;");
+                    GenType(ec);
+
+                    _context.AddSource($"{ec.ClassName}.Generated", SourceText.From(_builder.ToString(), Encoding.UTF8));
+
+                    _builder.Clear();
+                }
+            }
+
+            private void GenType(EventSourceClass ec)
+            {
+                if (!string.IsNullOrWhiteSpace(ec.Namespace))
+                {
+                    _builder.AppendLine($@"
+namespace {ec.Namespace}
+{{");
```

Comment:
Is the generator project always going to be built on the same platform on which it's run?  Verbatim strings end up including the newline of whatever platform the code is compiled on.

That asked, it probably doesn't matter, given the scenario.

Reply:
Yes; though its then compiled to il and the C# is dropped so shouldn't matter?